File: Python/Arrays/L_matrixDiagSum.py
def diagonalSum(mat) -> int:
        n = len(mat)
        diagnolSum = 0
        i,jl,jr = 0,0,n-1 
        while i < n:
            diagnolSum += mat[i][jl]
            diagnolSum += mat[i][jr]
            i+=1
            jl+=1
            jr-=1
        return diagnolSum if n%2==0 else diagnolSum - mat[n//2][n//2]

File: cpp/Array/L_matrixDiagSum.cpp
#include<bits/stdc++.h>
using namespace std;

int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size();
        int totsum = 0;
        int i=0,jl = 0, jr = n-1;
        while(i<n) {
            totsum += mat[i][jl];
            totsum += mat[i][jr];
            i++;
            jl++;
            jr--;
        }
        if(n%2) totsum -= mat[n/2][n/2];
        return totsum;
    }